frontend/src/components/SignUpForm/SignUpForm.jsx

```diff
@@ -0,0 +1,84 @@
+import "./SignUpForm.css";
+import { useState } from "react";
+import axios from "axios";
+import { Link } from "react-router-dom";
+
+export default function SignUpForm({ setUser }) {
+    const [username, setUsername] = useState("");
+    const [password, setPassword] = useState("");
+    const [email, setEmail] = useState("");
+
+    async function handleSubmit(event) {
+        event.preventDefault();
+
+        try {
+            const userData = {
+                username,
+                password,
+                email
+            };
+
+            const response = await axios.post(
+                "http://localhost:3000/users/signup",
+                userData,
+                { withCredentials: true }
+            );
+
+            setUsername("");
+            setPassword("");
+            setEmail("");
+
+            const user = response.data.user;
+            setUser(user);
+
+            alert("Sign Up Successful.");
```

Comment:
This is not always the case. Sometimes the response.data can bring something different than a successfully signed up user.